server/chatroom.py

```diff
@@ -35,21 +46,104 @@ def saveMessage(cur, sender_id, chatroomID, message_sent):
             {"sender_id": sender_id, "chatroomID": chatroomID, "message_sent": message_sent,
              "date_time_sent": date_time_sent})
         return True
-    except:
-        return False
+    except BaseException as e:
+        print(f'Exception: {e}')
+        raise HTTPException(
+            status_code=500,
+            detail="Database Error",
+        )
 
 
 # Create non-course chatroom
 def createChatroom(cur, user_id, chatroom_name, uni_id):
     try:
-        cur.execute("INSERT INTO chatroom(chatroom_name, uni_id) VALUES(%(chatroom_name)s, %(uni_id)s)",
-                    {"chatroom_name": chatroom_name, "uni_id": uni_id})
-        cur.execute("SELECT id FROM chatroom where chatroom_name = %(chatroom_name)s   AND uni_id = %(uni_id)s",
-                    {"chatroom_name": chatroom_name, "uni_id": uni_id})
+        # Generate invite id
+        invite_id = ''.join(random.choices(string.ascii_uppercase + string.digits, k=10))
+        
+        cur.execute("INSERT INTO chatroom(chatroom_name, uni_id, invite_id) VALUES(%(chatroom_name)s, %(uni_id)s, %(invite_id)s)", {"chatroom_name" : chatroom_name, "uni_id" : uni_id, "invite_id" : invite_id})
+        
+        cur.execute("SELECT id FROM chatroom where chatroom_name = %(chatroom_name)s   AND uni_id = %(uni_id)s", {"chatroom_name" : chatroom_name, "uni_id" : uni_id})
+        
         chatroom_id = cur.fetchall()[0][0]
+        
         cur.execute(
             "INSERT INTO in_chatroom(student_id, uni_id, chatroom_id) VALUES(%(user_id)s, %(uni_id)s, %(chatroom_id)s)",
             {"user_id": user_id, "uni_id": uni_id, "chatroom_id": chatroom_id})
+        
         return True
-    except:
-        return False
+    except BaseException as e:
+        print(f'Exception: {e}')
+        raise HTTPException(
+            status_code=500,
+            detail="Database Error",
+        )
+    
+#Generate invite link for student (for non-course chatroom)
+def GenerateInvite(cur, target_user_id, chatroom_id):
```

Comment:
nit: name python functions in this_form. No camelCase and definitely dont not capitalize the first letter. u should only capitalize the first letter for classes

Reply:
related to the comment i had in the sql file, we can also add code to enforce the non-course rule here if ur more comfortable with python rather than sql